client/p.go

```diff
@@ -808,21 +954,20 @@ func (pc *p) stake(k key.Key, fee uint64, opts ...OpOption) (
 
 		// add the input to the consumed inputs
 		ins = append(ins, in)
-		signers = append(signers, inputSigners...)
 	}
 
 	if amountStaked > 0 && amountStaked < ret.stakeAmt {
-		return nil, nil, nil, nil, ErrInsufficientBalanceForStakeAmount
+		return nil, nil, nil, ErrInsufficientBalanceForStakeAmount
 	}
 	if amountBurned > 0 && amountBurned < fee {
-		return nil, nil, nil, nil, ErrInsufficientBalanceForGasFee
+		return nil, nil, nil, ErrInsufficientBalanceForGasFee
 	}
 
-	avax.SortTransferableInputsWithSigners(ins, signers)      // sort inputs and keys
```

Comment:
Need to figure out how to keep signer addresses in same order

Reply:
```
type innerSortTransferableInputsWithSigners struct {
	ins     []*TransferableInput
	signers [][]*crypto.PrivateKeySECP256K1R
}

func (ins *innerSortTransferableInputsWithSigners) Less(i, j int) bool {
	iID, iIndex := ins.ins[i].InputSource()
	jID, jIndex := ins.ins[j].InputSource()

	switch bytes.Compare(iID[:], jID[:]) {
	case -1:
		return true
	case 0:
		return iIndex < jIndex
	default:
		return false
	}
}
func (ins *innerSortTransferableInputsWithSigners) Len() int { return len(ins.ins) }
func (ins *innerSortTransferableInputsWithSigners) Swap(i, j int) {
	ins.ins[j], ins.ins[i] = ins.ins[i], ins.ins[j]
	ins.signers[j], ins.signers[i] = ins.signers[i], ins.signers[j]
}

// SortTransferableInputsWithSigners sorts the inputs and signers based on the
// input's utxo ID
func SortTransferableInputsWithSigners(ins []*TransferableInput, signers [][]*crypto.PrivateKeySECP256K1R) {
	sort.Sort(&innerSortTransferableInputsWithSigners{ins: ins, signers: signers})
}
```

https://github.com/ava-labs/avalanchego/blob/224c9fd23d41839201dd0275ac864a845de6e93e/vms/components/avax/transferables.go#L176